### ISSUE TITLE
JIT - Channel Fee: LSP Flow

### ISSUE THREAD

**Issue by bonomat** [get10101/10101#843]
Follow-up ticket from #577 

LSP flow: there is work on Bolt X (TBD) which says that nodes may accept payments smaller than amounts in the invoice. Once this is done, we can subtract a channel fee from the payment amount same as Breez does. This is defined in [LSPS2](https://github.com/ZmnSCPxj-jr/lsp/blob/jit-channels/LSPS2/README.md?rgh-link-date=2023-05-02T21%3A21%3A21Z) of the upcoming LSP Spec and will find its way into bolts and all implementations soon. https://github.com/get10101/10101/issues/393

**Comment by bonomat** [comment]
Might work once we upgrade to 0.116 of LDK: https://github.com/lightningdevkit/rust-lightning/issues/1999
